### PR TITLE
Add mariadb swagger spec

### PR DESCRIPTION
**Pull request by elemount**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a few comments. Please refactor your models and apply readonly-ness as pointed out,
Do get a feeling this needs an API review.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 1]
```diff
@@ -0,0 +1,1929 @@
...
+        },
+        "storageMB": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Max storage allowed for a server."
+        }
+      },
+      "description": "Storage Profile properties of a server"
+    },
+    "ServerPropertiesForCreate": {
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: I set it to `ServerPropertiesForCreate` to concise with MySQL and PostgreSQL. For example, for java SDK, MySQL and PostgreSQL both use it as the API interface. We want to match with them.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 2]
```diff
@@ -0,0 +1,1929 @@
...
+          "description": "Resource name to verify."
+        },
+        "type": {
+          "type": "string",
+          "description": "Resource type used for verification."
+        }
+      },
+      "description": "Request from client to check resource name availability."
+    },
+    "NameAvailability": {
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: Agree with you. That is for matching with MySQL and PostgreSQL.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 3]
```diff
@@ -0,0 +1,1929 @@
...
+      "description": "A server security alert policy.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SecurityAlertPolicyProperties",
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: at line 1785 of same file.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 4]
```diff
@@ -0,0 +1,1929 @@
...
+    "SslEnforcement": {
+      "type": "string",
+      "description": "Enable ssl enforcement or not when connect to server.",
+      "enum": [
+        "Enabled",
+        "Disabled"
+      ],
+      "x-ms-enum": {
+        "name": "SslEnforcementEnum",
+        "modelAsString": false
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: That is expected. The value is a boolean originally. On reviewing MySQL and PostgreSQL, we were pointed out that we should use a enum for the enforce or not. This enum only have two option and will not add any options.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 5]
```diff
@@ -0,0 +1,1929 @@
...
+          "description": "Application-specific metadata in the form of key-value pairs."
+        }
+      },
+      "required": [
+        "properties",
+        "location"
+      ],
+      "description": "Represents a server to be created."
+    },
+    "ServerUpdateParameters": {
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: They are totally different. The option customer create database are not the same thing for their update the server resource. Move common properties can lead misunderstanding.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 6]
```diff
@@ -0,0 +1,1929 @@
...
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Represents a server."
+    },
+    "ServerForCreate": {
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: We have several server creating model, so create `ServerPropertiesForDefaultCreate`, `ServerPropertiesForRestore`, `ServerPropertiesForGeoRestore`. They are inherit from `ServerPropertiesForCreate`. To match such creating model (Which do not a things which the server response we get). We use `ServerForCreate` and `Server`

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: Also, we need match with MySQL and PostgreSQL for same api.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 7]
```diff
@@ -0,0 +1,1929 @@
...
+          "type": "string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "Last modified timestamp of the log file."
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of the log file."
+        },
+        "url": {
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: Sure, I've set it to readonly.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 8]
```diff
@@ -0,0 +1,1929 @@
...
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "TrackedResource": {
+      "description": "Resource properties including location and tags for track resources.",
+      "properties": {
+        "location": {
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: Sure.

File: specification/mariadb/resource-manager/readme.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+## Configuration
+
+
+
+### Basic Information
+These are the global settings for the Sql API.
+
+``` yaml
+openapi-type: arm
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/readme.md]: Thanks, I've updated it.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json [hunk 10]
```diff
@@ -0,0 +1,1929 @@
...
+    },
+    "Operation": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the operation being performed on this particular object."
+        },
+        "display": {
+          "$ref": "#/definitions/OperationDisplay",
```
[the remark this answers is not in the document]
Comment (elemount) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2018-06-01-preview/mariadb.json]: Sure.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**